packages/lib/src/interactions/SelectToZoom.tsx

```diff
@@ -3,19 +3,21 @@ import { useThree } from '@react-three/fiber';
 import { useCameraState } from '../vis/hooks';
 import { useAxisSystemContext } from '../vis/shared/AxisSystemProvider';
 import { getVisibleDomains } from '../vis/utils';
-import RatioSelectionRect from './RatioSelectionRect';
 import SelectionRect from './SelectionRect';
 import SelectionTool from './SelectionTool';
 import { useMoveCameraTo } from './hooks';
 import type { Interaction, Selection } from './models';
-import { getEnclosedRectangle, getRatioRespectingRectangle } from './utils';
+import { getEnclosedRectangle, getZoomTransform } from './utils';
 
 interface Props extends Omit<Interaction, 'button'> {
   keepRatio?: boolean;
+  fullDimIfEmpty?: boolean;
```

Comment:
Not a huge fan of the name - what about `withAxialZoom` or `fillIfEmpty`? Also, it doesn't make sense for both props to be `true`, so perhaps we should add a `throw` when this happens (as doing an XOR of two optional boolean props in TS is not trivial).